presto-main/src/main/java/com/facebook/presto/sql/gen/OrderingCompiler.java

```diff
@@ -234,6 +257,102 @@ private static void generateCompareTo(ClassDefinition classDefinition, CallSiteB
                 .retInt();
     }
 
+    public MergeSortComparator compileMergeSortComparator(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
+    {
+        requireNonNull(sortTypes, "sortTypes is null");
+        requireNonNull(sortChannels, "sortChannels is null");
+        requireNonNull(sortOrders, "sortOrders is null");
+
+        try {
+            return mergeSortComparators.get(new PagesIndexComparatorCacheKey(sortTypes, sortChannels, sortOrders));
+        }
+        catch (ExecutionException | UncheckedExecutionException | ExecutionError e) {
+            throw Throwables.propagate(e.getCause());
+        }
+    }
+
+    @VisibleForTesting
+    public MergeSortComparator internalCompileMergeSortComparator(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
+    {
+        MergeSortComparator comparator;
+        try {
+            Class<? extends MergeSortComparator> mergeSortComparatorClass = generateMergeSortComparatorClass(sortTypes, sortChannels, sortOrders);
+            comparator = mergeSortComparatorClass.newInstance();
+        }
+        catch (Throwable t) {
+            log.error(t, "Error compiling merge sort comparator for channels %s with order %s", sortChannels, sortChannels);
+            comparator = new SimpleMergeSortComparator(sortTypes, sortChannels, sortOrders);
```

Comment:
It seems like the same sort of case? We want to know when the bytecode generation fails so we can fix bugs in it. But I don't know enough about the bytecode to make an informed argument about it.